engine/src/test/java/com/datatorrent/stram/StramMiniClusterTest.java

```diff
@@ -165,22 +168,27 @@ public static void tearDown() throws IOException
     }
   }
 
-  @Test
-  public void testSetupShutdown() throws Exception
+  private void checkNodeState() throws YarnException
   {
-    GetClusterNodesRequest request =
-        Records.newRecord(GetClusterNodesRequest.class);
+    GetClusterNodesRequest request = Records.newRecord(GetClusterNodesRequest.class);
     ClientRMService clientRMService = yarnCluster.getResourceManager().getClientRMService();
     GetClusterNodesResponse response = clientRMService.getClusterNodes(request);
     List<NodeReport> nodeReports = response.getNodeReports();
     LOG.info("{}", nodeReports);
 
     for (NodeReport nr: nodeReports) {
-      LOG.info("Node: {}", nr.getNodeId());
-      LOG.info("Total memory: {}", nr.getCapability());
-      LOG.info("Used memory: {}", nr.getUsed());
-      LOG.info("Number containers: {}", nr.getNumContainers());
+      if (nr.getNodeState().isUnusable()) {
+        continue;
+      }
+      return;
     }
+    fail("Yarn Mini cluster should have at least one usable node.");
```

Comment:
The test would still fail when there is no usable node isn't it but will prevent it from blocking forever. Is my understanding correct?

Reply:
Yes, it will fail the test instead of blocking forever. Note that disabling disk health checker should help with making a node usable.